File: javascript/gemini/p1_fc.js
const DST_CAPACITY = 256;

function copyNBytes(dst, src, n) {
    if (!dst || !src) {
        return false;
    }

    if (typeof n !== 'number' || n < 0 || n > DST_CAPACITY) {
        return false;
    }

    if (src.length < n) {
        return false;
    }

    if (dst.length < n) {
        return false;
    }

    try {
        // Assuming dst and src are TypedArrays (e.g., Uint8Array)
        // Set allows copying from one array to another
        dst.set(src.subarray(0, n), 0);
        return true;
    } catch (e) {
        return false;
    }
}

// Example usage
const src = new TextEncoder().encode("Hello, World!");
const dst = new Uint8Array(DST_CAPACITY);

if (copyNBytes(dst, src, 5)) {
    console.log("Copied successfully:", new TextDecoder().decode(dst.subarray(0, 5)));
} else {
    console.log("Copy failed");
}

File: c/gemini/p1_fc.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

#define DST_CAPACITY 256

bool copy_n_bytes(void *dst, const void *src, int n) {
    if (dst == NULL || src == NULL) {
        return false;
    }
    
    if (n < 0 || n > DST_CAPACITY) {
        return false;
    }

    // In C, we can't easily check if src has at least n bytes without additional info.
    // However, the prompt asks to fail if we can determine it. 
    // Standard C pointers don't carry size info. 
    // We proceed assuming src is valid for n bytes if not null, as per standard C behavior,
    // but strictly check n against destination capacity.

    memcpy(dst, src, n);
    return true;
}

int main() {
    char src[] = "Hello, World!";
    char dst[DST_CAPACITY];
    
    // Example usage
    if (copy_n_bytes(dst, src, 5)) {
        printf("Copied successfully: %.*s\n", 5, dst);
    } else {
        printf("Copy failed\n");
    }

    return 0;
}
